### PR TITLE
Tax Rules Group : Edit Page > Tax Rules Grid (Only the list)

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Tax Rules Group : Edit Page > Tax Rules Grid (Only the list)
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | N/A
| How to test?      | Make assets and Go to this URI: `/admin-dev/index.php/improve/international/tax_rules_groups/`<br>Edit any tax rule group<br>You should see the form (migrated in this PR #28812)<br />You should see the list with all colums<br>No sort/filter is available like in the legacy page<br>No actions are available (Next PRs : Add/Edit Tax Rule & Delete/Bulk Delete Tax Rule)
| Fixed ticket?     | Relative to #10610
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

### CODE REVIEW

**Review by 0x346e3730 — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]
Nothing obviously wrong. Changes 0,1,2 values to redable things. Formats things properly in a decorator. Queries look OK. Default filter the same as in legacy. Proper way of dynamic applying filters to grid before presenting. Template is clean.

Franck for president :D

File: src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                    ->setOptions([
+                        'field' => 'description',
+                    ])
+            )
+            ->add(
+                (new ActionColumn('actions'))
+                    ->setName($this->trans('Actions', [], 'Admin.Global'))
+                    ->setOptions([
+                        'actions' => (new RowActionCollection())
+                        /*
```
Comment (zuk3975) [inline src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php:125]: would be nice to have a comment (preferable a todo reminder) of why these are commented out (I suspect its just expected to be done in separate PR).
Not blocking though (unless you have other feedback to fix).

File: src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getBulkActions()
+    {
+        return new BulkActionCollection()
+            /*
```
Comment (zuk3975) [inline src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php:177]: same here

File: src/Core/Grid/Query/TaxRuleQueryBuilder.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            'tr.`id_tax` = t.`id_tax`'
+        );
+
+        $qb->andWhere('tr.`id_tax_rules_group` = :idTaxRulesGroup');
+
+        $qb
+            ->setParameter('idLang', $this->employeeIdLang)
+            ->setParameter('idTaxRulesGroup', $filters['taxRulesGroupId']);
+
+        return $qb;
```
Comment (zuk3975) [inline src/Core/Grid/Query/TaxRuleQueryBuilder.php]: lots of unnecessary separations. It could all be done in one like:
```
        return $this->connection->createQueryBuilder()
            ->from($this->dbPrefix . 'tax_rule', 'tr')
            ->leftJoin(
                'tr',
                $this->dbPrefix . 'country',
                'c',
                'tr.`id_country` = c.`id_country`'
            )
            ->leftJoin(
                'tr',
                $this->dbPrefix . 'country_lang',
                'cl',
                'tr.`id_country` = cl.`id_country` AND cl.`id_lang` = :idLang '
            )
            .... etc
```

File: src/Core/Search/Filters/TaxRulesFilters.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace PrestaShop\PrestaShop\Core\Search\Filters;
+
+use PrestaShop\PrestaShop\Core\Grid\Definition\Factory\TaxRuleGridDefinitionFactory;
+use PrestaShop\PrestaShop\Core\Search\Filters;
+
+/**
+ * Provides default filters for tax rule grid.
+ */
+final class TaxRulesFilters extends Filters
```
Comment (zuk3975) [inline src/Core/Search/Filters/TaxRulesFilters.php]: as far as I see we use mostly singular form, would be nice to keep it that way. (not blocking though, because its not documented :man_shrugging: )
```suggestion
final class TaxRuleFilters extends Filters
```

File: src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml [hunk 5]
```diff
@@ -464,3 +464,10 @@ services:
       - '@prestashop.core.query.doctrine_search_criteria_applicator'
       - "@=service('prestashop.adapter.legacy.context').getContext().language.id"
     public: false
+
+  PrestaShop\PrestaShop\Core\Grid\Query\TaxRuleQueryBuilder:
+    parent: 'prestashop.core.grid.abstract_query_builder'
+    arguments:
+      - '@prestashop.core.query.doctrine_search_criteria_applicator'
+      - "@=service('prestashop.adapter.legacy.context').getContext().employee.id_lang"
+    public: true
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml]:
```suggestion
    public: false
```
unless it doesn't work with parent definition or so?

File: src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml [hunk 6]
```diff
@@ -367,3 +367,8 @@ services:
     class: 'PrestaShop\PrestaShop\Core\Grid\Definition\Factory\CustomerGroupsGridDefinitionFactory'
     parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
     public: true
+
+  PrestaShop\PrestaShop\Core\Grid\Definition\Factory\TaxRuleGridDefinitionFactory:
+    parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
+    public: true
```
Comment (zuk3975) [inline src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml]:
```suggestion
    public: false
```
same as above comment

**Comment by zuk3975** [comment]
nice PR overall, I just had some nitpick comments that I think is worth cleaning up

File: src/Core/Search/Filters/TaxRulesFilters.php (hunk 4 above)
Comment (Progi1984) [reply src/Core/Search/Filters/TaxRulesFilters.php]: Fixed

File: src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php (hunk 1 above)
Comment (Progi1984) [reply src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php:125]: It's my next PR.

File: src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php (hunk 2 above)
Comment (Progi1984) [reply src/Core/Grid/Definition/Factory/TaxRuleGridDefinitionFactory.php:177]: It's my next PR.

File: src/Core/Grid/Query/TaxRuleQueryBuilder.php (hunk 3 above)
Comment (Progi1984) [reply src/Core/Grid/Query/TaxRuleQueryBuilder.php]: Fixed

File: src/Core/Search/Filters/TaxRulesFilters.php (hunk 4 above)
Comment (zuk3975) [reply src/Core/Search/Filters/TaxRulesFilters.php]: either you forgot to push your changes or git is acting strange, but I don't see any changes after my last review

File: src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml (hunk 6 above)
Comment (Progi1984) [reply src/PrestaShopBundle/Resources/config/services/core/grid/grid_definition_factory.yml]: Fixed

**Comment by Progi1984** [comment]
> nice PR overall, I just had some nitpick comments that I think is worth cleaning up

@zuk3975 Done in 1d5e061

**Review by zuk3975 — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]

**Comment by Progi1984** [comment]
@zuk3975 & @Hlavtox I fixed the CI (Use of Constants in legacy Constants were in error for PHPStan).

**Review by zuk3975 — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Progi1984 ,

Tested the following:

Edit any tax rule group 
You should see the form (migrated in this PR https://github.com/PrestaShop/PrestaShop/pull/28812) 
You should see the list with all columns 
No sort/filter is available like in the legacy page
No actions are available (Next PRs : Add/Edit Tax Rule & Delete/Bulk Delete Tax Rule) 

Screenshot 2023-02-22 at 16 18 56

Waiting for the PR for :
Screenshot 2023-02-22 at 16 19 17


It is QA ✅ 
Thanks!

**Comment by Hlavtox** [comment]
Good job everyone!
